src/Terminus.php

```diff
@@ -293,8 +294,13 @@ private function configureContainer()
             ->invokeMethod('setSites', ['sites']);
 
         // Tell the command loader to only allow command functions that have a name/alias.
-        $factory = $container->get('commandFactory');
+        $command_cache = new FileStore($this->getConfig()->get('cache_dir') . '/commands');
+        $factory = new CachedAnnotatedCommandFactory();
+        $factory->setDataStore($command_cache);
         $factory->setIncludeAllPublicMethods(false);
+        $factory->setCommandProcessor($container->get('commandProcessor'));
+
+        $container->share('commandFactory', $factory);
```

Comment:
This use of `$container->get()` + `$container->share()` is dangerous. It works in the current instance because you are sharing only one object. If you followed this same pattern to fix up the container for several classes, then you could fall into a situation where `$container()->get()` might initialize one of the objects you have not fixed up yet. If you do this, the objects created before you alter the container won't be affected by the change to the container. This is fragile, and a trap for the unwary who might not understand what is going on.

It's better, then, to use `$container->share('commandFactory', CachedAnnotatedCommandFactory::class);`, and use the appropriate `withMethodCall()`s to insert references to `['commandProcessor']` instead of fetching and using $commandProcessor.

If these changes are pushed upstream, then fixing up the container won't be necessary, though.